src/dist.rs

```diff
@@ -43,10 +43,31 @@ pub(crate) mod rust {
     for (slice_org, slice_ref) in
       plane_org.rows_iter().take(h).zip(plane_ref.rows_iter())
     {
-      sum += slice_org
+      let mut iter_org = slice_org[..w].chunks_exact(4);
+      let mut iter_ref = slice_ref[..w].chunks_exact(4);
+      for chunk_org in iter_org.by_ref() {
+        let chunk_ref = iter_ref.next().expect("rows are same size");
```

Comment:
why not using zip?

Reply:
The first time I tried it the compiler was yelling at me. I went back again and it works, I think I was missing the `by_ref`